src/dndbeyond/content-scripts/character.js

```diff
@@ -1300,6 +1310,11 @@ function rollSpell(force_display = false, force_to_hit_only = false, force_damag
             damage_types.push(dmgtype);
         }
 
+        addCustomDamages(damages, damage_types);
+
+        // Capitalize all Damage Types to ensure consistency for later processing
+        damage_types = damage_types.map(t => capitalize(t));
+
```

Comment:
you moved the "addCustomDamages" from after special handling to before, which I think was done by mistake and could cause issues.